event_routing_backends/processors/caliper/transformer.py

```diff
@@ -39,17 +40,52 @@ def _add_generic_fields(self):
             'eventTime': convert_datetime_to_iso(self.get_data('timestamp', True)),
             'extensions': {}
         })
-        self.transformed_event['object'] = {
-            'extensions': {
-                'course_id': self.get_data('context.course_id')
-            }
-        }
+        self.transformed_event['object'] = {}
+        course_id = self.get_data('context.course_id')
+        if course_id is not None:
+            extensions = {"isPartOf": {}}
+            extensions['isPartOf']['id'] = self.get_object_iri('course', course_id)
+            extensions['isPartOf']['type'] = 'CourseOffering'
+            self.transformed_event['object']['extensions'] = {}
+            self.transformed_event['object']['extensions'].update(extensions)
 
     def _add_actor_info(self):
         """
         Add all generic information related to `actor`.
         """
         self.transformed_event['actor'] = {
-            'id': get_anonymous_user_id(self.extract_username_or_userid()),
+            'id': self.get_object_iri(
+                'user',
+                get_anonymous_user_id(self.extract_username_or_userid()),
+            ),
             'type': 'Person'
         }
+
+    def _add_generated_info(self):
```

Comment:
we need to move this logic in `ProblemEventsTransformers`
define `additional_fields= ('generated',)` property at the tope of `ProblemEventsTransformers`
and then write a method named `get_generated` like this
 ```suggestion
    def get_generated(self):
        return {
                'score': {
                    'id': '_:score',
                    'type': 'Score',
                    'maxScore': self.get_data('max_grade'),
                    'maxGiven': self.get_data('grade'),
                    'attempts': self.get_data('attempts'),
                    'extensions': {'success': "True" if self.get_data('success') == 'correct' else "False"},
                }
            }
```